src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1481,6 +1498,20 @@ lemma preconnected.subsingleton_connected_component (h : G.preconnected) :
   subsingleton G.connected_component :=
 ⟨connected_component.ind₂ (λ v w, connected_component.sound (h v w))⟩
 
+/-- The map on connected components induced by a graph homomorphism. -/
+def connected_component.map {V : Type*} {G : simple_graph V} {V' : Type*} {G' : simple_graph V'}
+  (φ : G →g G') (C : G.connected_component) : G'.connected_component :=
+C.lift (λ v, G'.connected_component_mk (φ v)) $ λ v w p _,
+  connected_component.eq.mpr (p.map φ).reachable
```

Comment:
We've already got the two functoriality lemmas, but I suppose we're missing the defining lemma, where for `v : V` we have `(G.connected_component_mk v).map φ = G'.connected_component_mk (φ v)`. That should be provable by `rfl`, but you can also use `connected_component.lift_mk`.

Naming-wise, I think this would be `connected_component.map_mk`.